command/keyring.go

```diff
@@ -1,37 +1,46 @@
 package command
 
 import (
-	"flag"
 	"fmt"
 	"strings"
 
+	consulapi "github.com/hashicorp/consul/api"
 	"github.com/hashicorp/consul/command/agent"
+	"github.com/hashicorp/consul/command/base"
 	"github.com/mitchellh/cli"
 )
 
 // KeyringCommand is a Command implementation that handles querying, installing,
 // and removing gossip encryption keys from a keyring.
 type KeyringCommand struct {
-	Ui cli.Ui
+	base.Command
 }
 
 func (c *KeyringCommand) Run(args []string) int {
-	var installKey, useKey, removeKey, token string
+	var installKey, useKey, removeKey string
 	var listKeys bool
 	var relay int
 
-	cmdFlags := flag.NewFlagSet("keys", flag.ContinueOnError)
-	cmdFlags.Usage = func() { c.Ui.Output(c.Help()) }
-
-	cmdFlags.StringVar(&installKey, "install", "", "install key")
-	cmdFlags.StringVar(&useKey, "use", "", "use key")
-	cmdFlags.StringVar(&removeKey, "remove", "", "remove key")
-	cmdFlags.BoolVar(&listKeys, "list", false, "list keys")
-	cmdFlags.StringVar(&token, "token", "", "acl token")
-	cmdFlags.IntVar(&relay, "relay-factor", 0, "relay factor")
-
-	rpcAddr := RPCAddrFlag(cmdFlags)
-	if err := cmdFlags.Parse(args); err != nil {
+	f := c.Command.NewFlagSet(c)
+
+	f.StringVar(&installKey, "install", "",
+		"Install a new encryption key. This will broadcast the new key to "+
+			"all members in the cluster.")
+	f.StringVar(&useKey, "use", "",
+		"Change the primary encryption key, which is used to encrypt "+
+			"messages. The key must already be installed before this operation "+
+			"can succeed.")
+	f.StringVar(&removeKey, "remove", "",
+		"Remove the given key from the cluster. This operation may only be "+
+			"performed on keys which are not currently the primary key.")
+	f.BoolVar(&listKeys, "list", false,
+		"List all keys currently in use within the cluster.")
+	f.IntVar(&relay, "relay-factor", 0,
+		"Added in Consul 0.7.4, setting this to a non-zero value will cause nodes "+
```

Comment:
I wonder if it makes sense to put these version notes in the actual command help, or just the web docs. If you are seeing this, then you have the right version :-)